libraries/gui/widgets/splash_screen.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * splash_screen.cpp
+ *
+ * Vega Strike - Space Simulation, Combat and Trading
+ * Copyright (C) 2001-2025 The Vega Strike Contributors:
+ * Project creator: Daniel Horn
+ * Original development team: As listed in the AUTHORS file
+ * Current development team: Roy Falk, Benjamen R. Meyer, Stephen G. Tuggy
+ *
+ *
+ * https://github.com/vegastrike/Vega-Strike-Engine-Source
+ *
+ * This file is part of Vega Strike.
+ *
+ * Vega Strike is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Vega Strike is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Vega Strike.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// -*- mode: c++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+
+#include <cassert>
+#include <string>
+#include <iostream>
+#include <SDL2/SDL.h>
+#include <boost/filesystem.hpp>
+
+
+
+#include "collections.h"
+#include "gui.h"
+
+// Must come before imgui.h
+#define IMGUI_DEFINE_MATH_OPERATORS
+#include "imgui.h"
+#include "backends/imgui_impl_sdl2.h"
+#include "backends/imgui_impl_opengl3.h"
+#include "backends/imgui_impl_sdlrenderer2.h"
+
+int (*displayTexturePtr)(int, int) = nullptr;
+
+static const ImGuiWindowFlags window_flags = 
+        ImGuiWindowFlags_NoTitleBar |
+        ImGuiWindowFlags_NoResize |
+        ImGuiWindowFlags_NoMove |
+        ImGuiWindowFlags_NoScrollbar |
+        ImGuiWindowFlags_NoCollapse;   // makes it transparent
+
+static ColorCollection colors;
+
+// Image to display out of std::vector<ImageData> splash_images 
+// in image_utils.cpp
+int splash_index = 0;   
+
+// A counter of times to run RenderSplashScreen before switching images.
+int splash_counter = 0;
+
+// Sets when to switch images.
+// TODO: move to configuration
+int splash_counter_max = 150;
+
+void RenderSplashScreen(const std::string message, int width, int height) {
+    if(!gui_initialized) {
+        return;
+    }
+
+    // This is used to generate a random name for the ImGui window,
+    // in order to prevent collisions.
+    const std::string* message_ptr = &message;
+    const std::string window_name = std::to_string((unsigned long long)message_ptr);
```

Comment:
```suggestion
    const std::string window_name = std::to_string(static_cast<uintmax_t>(message_ptr));
```

?

Reply:
This gives `invalid ‘static_cast’ from type ‘const std::string*’ {aka ‘const std::__cxx11::basic_string<char>*’} to type ‘uintmax_t’ {aka ‘long unsigned int’}`

How about:
```
const std::string* message_pointer = &message;
uintptr_t pointer_address = reinterpret_cast<uintptr_t>(message_pointer);    const std::string window_name = std::to_string(pointer_address);
```